mindsdb/integrations/handlers/elasticsearch_handler/elasticsearch_handler.py

```diff
@@ -1,142 +1,157 @@
 from mindsdb.integrations.libs.response import (
-    HandlerStatusResponse as StatusResponse,
     HandlerResponse as Response,
-    RESPONSE_TYPE
+    HandlerStatusResponse as StatusResponse,
+    RESPONSE_TYPE,
 )
-from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+from mindsdb.utilities import log
+
 
 logger = log.getLogger(__name__)
 
+
 class ElasticsearchHandler(DatabaseHandler):
     """
-    This handler handles connection and execution of the Airtable statements.
+    This handler handles the connection and execution of SQL statements on Elasticsearch.
     """
 
     name = 'elasticsearch'
 
-    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+    def __init__(self, name: Text, connection_data: Optional[Dict], **kwargs) -> None:
         """
-        Initialize the handler.
+        Initializes the handler.
+
         Args:
-            name (str): name of particular handler instance
-            connection_data (dict): parameters for connecting to the database
-            **kwargs: arbitrary keyword arguments.
+            name (Text): The name of the handler instance.
+            connection_data (Dict): The connection data required to connect to the AWS (S3) account.
+            kwargs: Arbitrary keyword arguments.
         """
         super().__init__(name)
-        self.parser = parse_sql
-        self.dialect = 'elasticsearch'
-
-        if ('hosts' not in connection_data) and ('cloud_id' not in connection_data):
-            raise Exception("Either the hosts or cloud_id parameter should be provided!")
-
-        optional_parameters = ['hosts', 'cloud_id', 'username', 'password']
-        for parameter in optional_parameters:
-            if parameter not in connection_data:
-                connection_data[parameter] = None
-
         self.connection_data = connection_data
         self.kwargs = kwargs
 
         self.connection = None
         self.is_connected = False
 
-    def __del__(self):
-        if self.is_connected is True:
+    def __del__(self) -> None:
+        """
+        Closes the connection when the handler instance is deleted.
+        """
+        if self.is_connected:
             self.disconnect()
 
-    def connect(self) -> StatusResponse:
+    def connect(self) -> Elasticsearch:
         """
-        Set up the connection required by the handler.
+        Establishes a connection to the Elasticsearch host.
+
+        Raises:
+            ValueError: If the expected connection parameters are not provided.
+
         Returns:
-            HandlerStatusResponse
+            elasticsearch.Elasticsearch: A connection object to the Elasticsearch host.
         """
-
         if self.is_connected is True:
-            return StatusResponse(True)
+            return self.connection
+
+        config = {}
+
+        # Mandatory connection parameters.
+        if ('hosts' not in self.connection_data) and ('cloud_id' not in self.connection_data):
```

Comment:
Shouldn't this be `or` ?

Reply:
I believe it should be and, @ZoranPandovski.
You can either provide `hosts` or the `cloud_id` parameter and then choose either `api_key` or `user` and `password` authentication.